.github/workflows/shopify-cli.yml

```diff
@@ -101,6 +101,31 @@ jobs:
       - name: Acceptance tests
         if: ${{ matrix.node == '18.7.0' }}
         run: pnpm nx run features:test
+      - name: Unit tests with coverage
+        if: ${{ matrix.node == '18.7.0' && matrix.os == 'ubuntu-latest' }}
+        run: pnpm nx run-many --all --skip-nx-cache --target=test:coverage --exclude=features
+      - name: Combine Jest coverages
+        run: npx istanbul-merge --out="coverage.raw.json" "packages/*/coverage/coverage-final.json"
+        if: ${{ matrix.node == '18.7.0' && matrix.os == 'ubuntu-latest' }}
+      - name: Convert coverage to Jest
+        run: ./bin/save-coverage-file.js
+        if: ${{ matrix.node == '18.7.0' && matrix.os == 'ubuntu-latest' }}
+      - name: Take copy of report as new baseline for branch
+        run: cp ./report.json ./baseline-report.json
+        if: ${{ matrix.node == '18.7.0' && matrix.os == 'ubuntu-latest' }}
+      - name: Clean ref name
+        if: ${{ matrix.node == '18.7.0' && matrix.os == 'ubuntu-latest' }}
+        env:
+            SAFE_REF_NAME: "${{ github.head_ref }}"
+        run: |
+          SAFE_REF_NAME=${{ env.SAFE_REF_NAME }}
+          SAFE_REF_NAME=${SAFE_REF_NAME//[\/.]/}
+          echo SAFE_REF_NAME=${SAFE_REF_NAME} >> $GITHUB_ENV
+      - uses: actions/upload-artifact@v3
+        if: ${{ matrix.node == '18.7.0' && matrix.os == 'ubuntu-latest' }}
+        with:
+          name: ${{ env.SAFE_REF_NAME }}--coverage-report
+          path: ./baseline-report.json
```

Comment:
could we extract the shared bits in a shared action inside `.github/actions` to reduce duplication? in this way we can also have a single `if: https://shopify.slack.com/archives/C0BUY1F5L/p1670411369252219`